src/provider/provider.py

```diff
@@ -0,0 +1,92 @@
+from abc import abstractmethod
+from numbers import Number
+
+
+class AbstractProvider:
```

Comment:
The methods on this class are conflicting with each other. You'll need to use default arguments instead of the overload pattern here. 

So, we should have only a single abstract method for `get_boolean_value` for example: 

```python
def get_boolean_value(
    self, 
    key: str, 
    default_value: bool, 
    evaluation_context: typing.Any = None, 
    flag_evaluation_options: typing.Any = None
) -> bool:
    ...
```

Reply:
Removed the polymorphic versions of the methods